apis/externalsecrets/v1beta1/secretstore_gcpsm_types.go

```diff
@@ -40,13 +40,14 @@ type GCPWorkloadIdentity struct {
 
 // GCPSMProvider Configures a store to sync secrets using the GCP Secret Manager provider.
 type GCPSMProvider struct {
-	// Auth defines the information necessary to authenticate against GCP
-	// +optional
-	Auth GCPSMAuth `json:"auth,omitempty"`
-
 	// ProjectID project where secret is located
-	ProjectID string `json:"projectID,omitempty"`
```

Comment:
This would be a breaking change for any existing environments that don't set this. I can see that in certain cases we error without a project id as is, but I'm not a 100% sure it's a required value in all situations.

Reply:
I will revert it and try to implement an autodetection of the project provided by the credentials. If I fail I will just leave it as it is.

Reply:
Currently in the official release when you try to add a secretstore or clustersecretstore without a projectID the status of this resource will report an "InvalidProviderConfig" due to the following line:

https://github.com/external-secrets/external-secrets/blob/325cc52790aa10312c6a8ed701d9e2f7d7cabee9/pkg/provider/gcp/secretmanager/provider.go#L149

So I will revert my changes as it works as intended already